src/Misc/HighDynamicRange/cubemapToSphericalPolynomial.ts

```diff
@@ -116,6 +116,10 @@ export class CubeMapToSphericalPolynomialTools {
         // The (u,v) of the first texel is half a texel from the corner (-1,-1).
         var minUV = du * 0.5 - 1.0;
 
+        // Compute the area on the unit sphere of the rectangle defined by (x,y) and the origin
+        // See https://www.rorydriscoll.com/2012/01/15/cubemap-texel-solid-angle/
+        var AreaElement = (x : number, y : number) => { return Math.atan2(x * y, Math.sqrt(x * x + y * y + 1)); };

```

Comment:
Would be great to move this into its own private function to prevent over allocation and garbage despite not being called frequently ?